File: cpp/test/TestTokenEnum.cpp

#include "TokenEnumRegistration.h"

#include "gtest/gtest.h"


TEST(TokenEnumTest, Trivia) {
    
    EXPECT_TRUE(TOKEN_WHITESPACE.isTrivia());
}

TEST(TokenEnumTest, PossibleBeginning) {
    
    EXPECT_TRUE(TOKEN_SYMBOL.isPossibleBeginning());
    
    EXPECT_TRUE(TOKEN_SEMISEMI.isPossibleBeginning());
}

TEST(TokenEnumTest, Closer) {
    
    EXPECT_TRUE(TOKEN_CLOSESQUARE.isCloser());
    
    EXPECT_TRUE(TOKEN_LONGNAME_RIGHTCEILING.isCloser());
}

TEST(TokenEnumTest, Error) {
    
    EXPECT_TRUE(TOKEN_ERROR_EXPECTEDTAG.isError());
    
    EXPECT_TRUE(TOKEN_ERROR_UNSUPPORTEDTOKEN.isError());
}

TEST(TokenEnumTest, Unterminated) {
    
    EXPECT_TRUE(TOKEN_ERROR_UNTERMINATEDSTRING.isUnterminated());
    
    EXPECT_TRUE(TOKEN_ERROR_UNTERMINATEDCOMMENT.isUnterminated());
}

TEST(TokenEnumTest, Empty) {
    
    EXPECT_TRUE(TOKEN_ENDOFFILE.isEmpty());
    
    EXPECT_TRUE(TOKEN_ERROR_EXPECTEDOPERAND.isEmpty());
}

File: crate/src/tests/test_token_enum.rs
use crate::token_enum::TokenEnum::*;


#[test]
fn TokenEnumTest_Trivia() {
    assert!(TOKEN_WHITESPACE.isTrivia());
}

#[test]
fn TokenEnumTest_PossibleBeginning() {
    assert!(TOKEN_SYMBOL.isPossibleBeginning());

    assert!(TOKEN_SEMISEMI.isPossibleBeginning());
}

#[test]
fn TokenEnumTest_Closer() {
    assert!(TOKEN_CLOSESQUARE.isCloser());

    assert!(TOKEN_LONGNAME_RIGHTCEILING.isCloser());
}

#[test]
fn TokenEnumTest_Error() {
    assert!(TOKEN_ERROR_EXPECTEDTAG.isError());

    assert!(TOKEN_ERROR_UNSUPPORTEDTOKEN.isError());
}

#[test]
fn TokenEnumTest_Unterminated() {
    assert!(TOKEN_ERROR_UNTERMINATEDSTRING.isUnterminated());

    assert!(TOKEN_ERROR_UNTERMINATEDCOMMENT.isUnterminated());
}

#[test]
fn TokenEnumTest_Empty() {
    assert!(TOKEN_ENDOFFILE.isEmpty());

    assert!(TOKEN_ERROR_EXPECTEDOPERAND.isEmpty());
}
